x-pack/plugins/apm/server/lib/rum_client/has_rum_data.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Setup, SetupTimeRange } from '../helpers/setup_request';
+import {
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+} from '../../../common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { rangeFilter } from '../../../common/utils/range_filter';
+import { TRANSACTION_PAGE_LOAD } from '../../../common/transaction_types';
+
+export async function hasRumData({ setup }: { setup: Setup & SetupTimeRange }) {
+  try {
+    const { start, end } = setup;
+
+    const params = {
+      apm: {
+        events: [ProcessorEvent.transaction],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [{ term: { [TRANSACTION_TYPE]: TRANSACTION_PAGE_LOAD } }],
+          },
+        },
+        aggs: {
+          services: {
+            filter: {
+              range: rangeFilter(start, end),
+            },
+            aggs: {
+              mostTraffic: {
+                terms: {
+                  field: SERVICE_NAME,
+                  size: 1,
+                },
+              },
+            },
+          },
+        },
+      },
+    };
```

Comment:
Just for curiosity, you could avoid passing the time range to the `hasData` if you'd have something like
```
//has_data.ts
GET apm-*/_search?terminate_after=1
{
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "transaction.type": "page-load"
          }
        },
        {
          "terms": {
            "processor.event": [
              "transaction"
            ]
          }
        },
        {
          "range": {
            "observer.version_major": {
              "gte": 7
            }
          }
        }
      ]
    }
  }
}
```

Then the API the fetches data:
```
//get_web_core_vitals.ts
 {
  query: {
    bool: {
      filter: [
        {
          range: {
            '@timestamp': {
              gte: 1601887365767,
              lte: 1601888265767,
              format: 'epoch_millis',
            },
          },
        },
        { term: { 'transaction.type': 'page-load' } },
        { exists: { field: 'transaction.marks.navigationTiming.fetchStart' } },
        { terms: { 'service.name': ['elastic-co-frontend'] } },
        { term: { 'user_agent.name': 'Chrome' } },
        { terms: { 'processor.event': ['transaction'] } },
      ],
    },
  },
  size: 0,
  aggs: {
    services: {
      terms: {
        field: 'service.name',
        size: 10,
      },
      aggs: {
        lcp: {
          percentiles: {
            field: 'transaction.marks.agent.largestContentfulPaint',
            percents: [50],
          },
        },
        fid: {
          percentiles: {
            field: 'transaction.experience.fid',
            percents: [50],
          },
        },
        cls: {
          percentiles: {
            field: 'transaction.experience.cls',
            percents: [50],
          },
        },
        tbt: {
          percentiles: {
            field: 'transaction.experience.tbt',
            percents: [50],
          },
        },
        fcp: {
          percentiles: {
            field: 'transaction.marks.agent.firstContentfulPaint',
            percents: [50],
          },
        },
        lcpRanks: {
          percentile_ranks: {
            field: 'transaction.marks.agent.largestContentfulPaint',
            values: [2500, 4000],
            keyed: false,
          },
        },
        fidRanks: {
          percentile_ranks: {
            field: 'transaction.experience.fid',
            values: [100, 300],
            keyed: false,
          },
        },
        clsRanks: {
          percentile_ranks: {
            field: 'transaction.experience.cls',
            values: [0.1, 0.25],
            keyed: false,
          },
        },
      },
    },
  },
};
```

With this small change, you wouldn't have to change the `hasData` signature and it wouldn't change anything in the performance of the page, since you'd still do two requests. Of course, I'm not 100% sure about the business logic you have in this section.  @sqren would you have some thoughts to share about this?

Reply:
yes this can be done but  whole part will become conditional, because this is also used on user experience app, so these aggs will be if service name is provided then use it as filter and only do web core vital aggs other wise do web core vitals aggs as childs aggs of parent service name aggs.

   ```
services: {
      terms: {
        field: 'service.name',
        size: 10,
      },
```

Reply:
but i will insist passing date range to hasRum data is pretty ok, someone else creating new has data api and passing date range as well, risk is far too low to consider refactor. 

on other hand changing web core vital app will disrupt other parts of the code. that change is significant.